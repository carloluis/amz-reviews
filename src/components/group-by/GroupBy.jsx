// Not implemented yet!

import React from 'react';
import Select from '../select/Select';

const GROUPS = ['', 'Day', 'Week', 'Month'];

const GroupBy = () => <Select onSelect={() => {}} title="Group by" values={GROUPS} disabled />;

export default GroupBy;
