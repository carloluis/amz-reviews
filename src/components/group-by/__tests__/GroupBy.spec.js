import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import GroupBy from '../GroupBy';

const PROPS = {
    groupBy: jest.fn(),
    value: 'some'
};

describe('<GroupBy/>', () => {
    it('snapshot', () => {
        const result = renderer.create(<GroupBy {...PROPS} />).toJSON();
        expect(result).toMatchSnapshot();
    });
});

describe('GroupBy', () => {
    it('should call the groupBy callback when selecting new value', () => {
        const wrapper = shallow(<GroupBy {...PROPS} />);
        const value = 'Day';
        wrapper.simulate('select', value);
        expect(PROPS.groupBy).toHaveBeenCalledWith(value);
    });
});
