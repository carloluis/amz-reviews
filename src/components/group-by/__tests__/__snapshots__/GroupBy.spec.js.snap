// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupBy/> snapshot 1`] = `
<div
  className="container"
>
  <div
    className="input"
    onClick={[Function]}
  >
    some
     
    <i
      className="triangle"
    >
      â–¼
    </i>
  </div>
</div>
`;
